opam-ci-check/lib/opam_build.ml

```diff
@@ -122,10 +115,9 @@ let setup_repository ?(local=false) ~variant ~for_docker ~opam_version () =
      Otherwise the "opam pin" after the "opam repository set-url" will fail (cannot find the new package for some reason) *)
   run "%s -f %s/bin/opam-%s %s/bin/opam" ln prefix opam_version_str prefix ::
   run ~network "opam init --reinit%s -ni" opamrc :: (* TODO: Remove ~network when https://github.com/ocurrent/ocaml-dockerfile/pull/132 is merged *)
-  run "uname -rs && opam exec -- ocaml -version && opam --version" ::
+  run "opam option solver=builtin-0install && opam config report" ::
```

Comment:
We tried enabling this in tests before, in https://github.com/ocurrent/opam-repo-ci/pull/351, which produced the regression in https://github.com/ocurrent/opam-repo-ci/issues/352

The problem was with the tests with earlier opam versions we are testing in https://github.com/ocurrent/opam-repo-ci/blob/e682604cd3f890291107f724d05e9e3896ed5be1/lib/build.ml#L156-L159

The conclusion we reached at that time was https://github.com/ocurrent/opam-repo-ci/issues/352#issuecomment-2332085848 . There may be some other useful context. See also https://github.com/ocurrent/opam-repo-ci/issues/354 .

AFACT, there is nothing in in this change that will avoid the problems with hit with opam 2.2, or is there and I am missing it?

So this makes me wonder if the solver timeouts may be a reason for us to try to accelerate https://github.com/ocurrent/docker-base-images/issues/132 and dropping support for opam 2.0 and 2.1 here...

I know we have wanted to keep testing on opam versions so long as they are in the repos of maintained distros, but if the quality of the tests are degrading across the board as a result, I think it may be counter productive. This just follows your compelling logic here, I think.

WDYT?